### DISCUSSION TITLE
File name or extension too long in PowerShell when trying to render Quarto manuscript

### DISCUSSION THREAD

**Discussion by Harbour-N** [quarto-dev/quarto-cli#9713] [Q&A]
### Description

I am writing my first manuscript in Quarto (unfortunately, I can't share it as it contains sensitive data), it was all working brilliantly until I started to get an error, when I try and run `quarto render`: 

```
ERROR: Error executing 'PowerShell': The filename or extension is too long. (os error 206)

Stack trace:
    at execProcess (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:5297:15)
    at zip (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:93248:12)
    at createMecaBundle (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:93563:33)
    at eventLoopTick (ext:core/01_core.js:183:11)
    at async Object.postRender (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:94336:32)
    at async renderProject (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:77614:13)
    at async Command.fn (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:81959:32)
    at async Command.execute (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:8104:13)
    at async quarto (file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:115002:5)
    at async file:///C:/Users/nicho/AppData/Local/Programs/Quarto/bin/quarto.js:115020:9 

```
I am on windows 11 and have enabled long paths but still get this error. I am not really sure why I keep getting this error. 

I have copy and pasted the manuscript folder and removed the .git file and it then rendered fine (I am not sure why this is).

Any suggestions would be appreciated.

**Comment by cscheid** [comment]
Unfortunately, we can't help you without a reproducible example. You don't need to share the example with sensitive data, but you can try to create a similar example that doesn't have sensitive data and still triggers the bug.

**Comment by cderv** [comment]
> I am on windows 11 and have enabled long paths but still get this error. I am not really sure why I keep getting this error.

This errors comes from our Powershell call use to zip the files in the meca bundle. We use `PowerShell Compress-Archive` on all the path to include in the archive folder. 

Just to be on the same page, we create the MECA bundle in specific case: https://quarto.org/docs/manuscripts/components.html#meca-bundle

Did you do this activation for Long File Path ? 
https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=powershell#enable-long-paths-in-windows-10-version-1607-and-later

> I have copy and pasted the manuscript folder and removed the .git file and it then rendered fine (I am not sure why this is).

This is suprising. I'll try to check if we somehow do not ignore the hidden .git folder. I can't reproduce this, so I am surprised that removing the git repo solved it. However, you also copy pasted the folder to another location, and maybe `.git` deletion does not matter - you just reduced the file path. 

The MAX_PATH limit is well known on Windows, and not all the tooling may support Long File Path. So maybe compress-archive does not. 
If you can build an example similar to yours, then I could try get it and put it in a place with long file path to see. 
Maybe we need to change the methods used, or this is just a windows limit unfortunately. 

> I am writing my first manuscript in Quarto (unfortunately, I can't share it as it contains sensitive data),

Can you give us at least some insigths on where this project is as a Path on your system ? Just to check how long this path is. Some tools helps to detect those problematic file path too

**Reply by Harbour-N** [reply]
> Did you do this activation for Long File Path ?
> https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=powershell#enable-long-paths-in-windows-10-version-1607-and-later

Yes I followed these steps to activate long path.

> This is suprising. I'll try to check if we somehow do not ignore the hidden .git folder. I can't reproduce this, so I am surprised that removing the git repo solved it. However, you also copy pasted the folder to another location, and maybe `.git` deletion does not matter - you just reduced the file path.

When I copy and pasted it was to the same directory. I have also been unable to reproduce this error in any other example. I have noticed the error has also gone away when I have made only changes to the index.qmd file and/or the notebooks, and not changed the file path and I am unsure why this might be.

> Can you give us at least some insigths on where this project is as a Path on your system ? Just to check how long this path is. Some tools helps to detect those problematic file path too

the file path is something like: C:\Users\nicho\OneDrive - The University of Nottingham\Desktop\quart\project . I have tried moving the folder to the root of my directory (C drive) to reduce the path length but that didn't work. I have a couple of other manuscript projects in the same directory and all them have no issues.

**Reply by cderv** [reply]
> the file path is something like: C:\Users\nicho\OneDrive - The University of Nottingham\Desktop\quart\project . I have tried moving the folder to the root of my directory (C drive) to reduce the path length but that didn't work. I have a couple of other manuscript projects in the same directory and all them have no issues.

This looks quite specific to this project. I don't really know what to look for in our code base to check for potential issue unfortunately. 

I see you are mentioning One Drive - we know the stack of tech we use can have issue on Remote Synced Drive, though usually it is a copy error during the rendering, not a powershell error. 

As I said, unfortunately, I don't know where to look for now. We'll keep that open in case other encounter similar issue. If you have any new information you can provide, they will be very welcome ! Thanks

**Comment by JackyFlint** [comment]
LongPathTool fixed long path errors that were blocking Quarto from rendering documents on Windows

**Reply by mcanouil** [reply]
What is that? Could you add context?
